src/index.ts

```diff
@@ -14,49 +14,60 @@ import type Koa from 'koa';
 // the `koa-bodyparser` types "polyfill" the `koa` types
 import type * as _ from 'koa-bodyparser';
 
-export interface KoaContextFunctionArgument {
-  ctx: Koa.Context;
+export interface KoaContextFunctionArgument<
+  StateT = Koa.DefaultState,
+  ContextT = Koa.DefaultContext,
+> {
+  ctx: Koa.ParameterizedContext<StateT, ContextT>;
 }
 
-interface KoaMiddlewareOptions<TContext extends BaseContext> {
-  context?: ContextFunction<[KoaContextFunctionArgument], TContext>;
+interface KoaMiddlewareOptions<TContext extends BaseContext, StateT, ContextT> {
```

Comment:
Just noting that this interface isn't exported and its only usage provides a defaulted `StateT` and `ContextT`.

This _does_ seems like it could be exported, in which case we'd probably want to default all 3 of these type params. If you care to make that change, maybe in a separate PR, but we can also just wait until somebody comes along and requests that addition.